app/screens/more_channels/index.js

```diff
@@ -17,30 +15,17 @@ import {isAdmin, isSystemAdmin} from '@mm-redux/utils/user_utils';
 import {getTheme} from '@mm-redux/selectors/entities/preferences';
 import {getConfig, getLicense} from '@mm-redux/selectors/entities/general';
 
-import MoreChannels from './more_channels';
-
-const joinablePublicChannels = createSelector(
-    getChannelsInCurrentTeam,
-    getMyChannelMemberships,
-    (channels, myMembers) => {
-        return channels.filter((c) => {
-            return (!myMembers[c.id] && c.type === General.OPEN_CHANNEL && c.delete_at === 0);
-        });
-    },
-);
+import {teamArchivedChannels, joinablePublicChannels, joinableSharedChannels} from '@selectors/channel';
 
-const teamArchivedChannels = createSelector(
-    getChannelsInCurrentTeam,
-    (channels) => {
-        return channels.filter((c) => c.delete_at !== 0);
-    },
-);
+import MoreChannels from './more_channels';
 
 function mapStateToProps(state) {
     const config = getConfig(state);
     const license = getLicense(state);
+    const sharedChannelsEnabled = config.ExperimentalSharedChannels === 'true';
     const roles = getCurrentUserRoles(state);
     const channels = joinablePublicChannels(state);
+    const sharedChannels = sharedChannelsEnabled ? joinableSharedChannels(state) : [];
```

Comment:
think of moving the empty array outside the function as a constant so it always references the same array instead of a new one every time this function gets called. by doing that you'll avoid the component to re-render without need.